nima/sse/webclient/src/main/java/io/helidon/nima/sse/webclient/SseSource.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.nima.sse.webclient;
+
+import io.helidon.common.GenericType;
+import io.helidon.nima.sse.SseEvent;
+import io.helidon.nima.webclient.http.spi.Source;
+
+/**
+ * A source for {@link SseEvent}s.
+ */
+public interface SseSource extends Source<SseEvent> {
+
+    /**
+     * A type representing an SSE source.
+     */
+    GenericType<SseSource> TYPE = GenericType.create(SseSource.class);
+
+    @Override
+    default void onEvent(SseEvent event) {
+    }
+}
```

Comment:
Does `onEvent` need to have noop default? Could `SseSource` be `@FunctionalInterface` so something like this is possible?
```java
r.source(SseSource.TYPE, (SseEvent event) -> System.out.println(event.data()));
```

Reply:
Yes, changed to functional interface.